src/sam/readrecord.jl

```diff
@@ -198,18 +198,12 @@ const sam_actions_header = merge(
     Dict(
         :countline => :(linenum += 1),
         :metainfo => quote
+            finish_header = true
             $(action_metainfo)
             push!(header, metainfo)
             metainfo = MetaInfo()
         end,
         :header => quote
-
-            finish_header = true
-
-            if !eof(stream)
-                p -= 1 # cancel look-ahead
```

Comment:
Did we determine whether this look ahead could be replaced by `Automa.where`?

Reply:
No, I didn't check that. Not even sure exactly what `Automa.where` does. The lookahead is an ugly solution - this PR makes the trailing newline a part of the header, such that it functions as a lookahead (but is a little more elegant).

It's still not that good. I've found it hard to make a regex that is both unambiguous, and which allows parsing a header without any trailing characters (is that even acutally needed)